pallets/asset-manager/src/lib.rs

```diff
@@ -283,6 +312,14 @@ pub mod pallet {
 			LocationAssetId::<T>::remove(&old_location);
 			LocationAssetId::<T>::insert(&location, &asset_id);
 			AssetIdLocation::<T>::insert(&asset_id, &location);
+
+			// extract para id from multilocation and store it.
+			if let Some(para_id) =
+				Self::get_para_id_from_multilocation(location.clone().into().as_ref())
+			{
+				AllowedDestParaIds::<T>::insert(para_id, ());
+			}
+
```

Comment:
what if this update removes that last asset of a certain parachain so that you need to remove the parachain id from the map?